cmd/containerd-shim-runhcs-v1/delete.go

```diff
@@ -18,6 +17,29 @@ import (
 	"go.opencensus.io/trace"
 )
 
+// LimitedRead reads at max `readLimitBytes` bytes from the file at path `filePath`. If the file has
+// more than `readLimitBytes` bytes of data then first `readLimitBytes` will be returned.
+func limitedRead(filePath string, readLimitBytes int64) ([]byte, error) {
+	f, err := os.Open(filePath)
+	if err != nil {
+		return nil, errors.New(fmt.Sprintf("limited read failed to open file: %s", err))
+	}
+	defer f.Close()
+	if fi, err := f.Stat(); err == nil {
+		if fi.Size() < readLimitBytes {
+			readLimitBytes = fi.Size()
+		}
+		buf := make([]byte, readLimitBytes)
+		_, err := f.Read(buf)
+		if err != nil {
+			return []byte{}, errors.New(fmt.Sprintf("limited read failed during file read: %s", err))
+		}
+		return buf, nil
+	} else {
```

Comment:
Nit: Don't need the else block, can just return the empty slice and error below the if case.